File: solutions/go/darts/1/darts.go
package darts

import (
	"math"
)

func Score(x, y float64) int {

	s := math.Sqrt(x*x + y*y)

	switch {
	case s > 10:
		return 0
	case s > 5:
		return 1
	case s > 1:
		return 5
	default:
		return 10
	}
}

File: solutions/c/darts/1/darts.c
#include "darts.h"
#include <math.h>

int score(coordinate_t landing_position) {
    float distance = sqrt(landing_position.x * landing_position.x + landing_position.y * landing_position.y);
    
    if (distance <= 1.0f) {
        return 10;
    } else if (distance <= 5.0f) {
        return 5;
    } else if (distance <= 10.0f) {
        return 1;
    }
    return 0;
}